tests/test_activator.py

```diff
@@ -0,0 +1,312 @@
+        # Second snap ignored
+        self.assertEqual(known_ids, {2025092600042})
+        self.ingester.assert_called_once_with(oids[0])
+
+    def test_two_ingest(self):
+        known_ids = set()
+        oids = [os.environ["RUBIN_INSTRUMENT"] + f"/20250926/blah_blah_20250926_{seq}"
+                f"/blah_blah_20250926_{seq}_R00_S00.fits"
+                for seq in [42, 43]]
+        with unittest.mock.patch("activator.activator.check_for_snap", side_effect=oids), \
+                unittest.mock.patch("activator.activator._get_storage_client"):
+            _ingest_existing_raws(self.visit, 2, self.ingester, known_ids)
+        self.assertEqual(known_ids, {2025092600042, 2025092600043})
+        self.assertEqual(self.ingester.call_count, 2)
+        for oid in oids:
+            self.ingester.assert_any_call(oid)
+
+    def test_two_ingest_duplicate(self):
+        known_ids = {2025092600042}
+        oids = [os.environ["RUBIN_INSTRUMENT"] + f"/20250926/blah_blah_20250926_{seq}"
+                f"/blah_blah_20250926_{seq}_R00_S00.fits"
+                for seq in [42, 43]]
+        with unittest.mock.patch("activator.activator.check_for_snap", side_effect=oids), \
+                unittest.mock.patch("activator.activator._get_storage_client"):
+            _ingest_existing_raws(self.visit, 2, self.ingester, known_ids)
+        self.assertEqual(known_ids, {2025092600042, 2025092600043})
+        self.ingester.assert_called_once_with(oids[1])
+
+
+class IsProcessableTest(unittest.TestCase):
+    def setUp(self):
+        now = astropy.time.Time.now().replicate(format="unix_tai")
+        # Generated 10 minutes before now, observed 8 minutes before
+        self.visit = FannedOutVisit(instrument=os.environ["RUBIN_INSTRUMENT"],
+                                    detector=90,
+                                    groupId="88",
+                                    nimages=1,
+                                    filters="k0123",
+                                    coordinateSystem=FannedOutVisit.CoordSys.ICRS,
+                                    position=[-42.0, 38.2],
+                                    startTime=float(now.value) - 500.0,
+                                    rotationSystem=FannedOutVisit.RotSys.SKY,
+                                    cameraAngle=0,
+                                    survey="SURVEY",
+                                    salIndex=3,
+                                    scriptSalIndex=3,
+                                    dome=FannedOutVisit.Dome.OPEN,
+                                    duration=35.0,
+                                    totalCheckpoints=1,
+                                    private_sndStamp=float(now.value) - 600.0,
+                                    )
+
+    def test_long_expiration(self):
+        self.assertTrue(is_processable(self.visit, 3600.0))
+        self.assertTrue(is_processable(self.visit, 610.0))  # Test may have long delays in Jenkins
```

Comment:
Does this mean that we'll see whether Jenkins has a long delay and fail this test?

Reply:
No, other way around: I gave the test a huge margin on that side so that people wouldn't complain if the nightly integration tests failed. See #343 and [this Slack thread](https://rubin-obs.slack.com/archives/C07QGMAN6RL/p1755240688153089).

Would something like "Pad for possible Jenkins runtime" be clearer?

Reply:
Thought comparing to #343 (which added 0.1 s), maybe I made the margin a little too big?

Reply:
Got it.  Thanks for explaining. Yes, "Pad for possible slow run time" would be clearer.